packages/lambda-tiler/src/index.ts

```diff
@@ -1,59 +1,52 @@
-import { lf, LambdaHttpRequest, LambdaHttpResponse } from '@linzjs/lambda';
 import { LogConfig } from '@basemaps/shared';
+import { LambdaHttpRequest, LambdaHttpResponse, lf } from '@linzjs/lambda';
+import { createHash } from 'crypto';
+import { Router } from './router.js';
 import { Ping, Version } from './routes/api.js';
+import { fontGet, fontList } from './routes/fonts.js';
 import { Health } from './routes/health.js';
+import { imageryGet } from './routes/imagery.js';
+import { spriteGet } from './routes/sprites.js';
 import { Tiles } from './routes/tile.js';
-import { Router } from './router.js';
-import { createHash } from 'crypto';
-import { Imagery } from './routes/imagery.js';
-import { Esri } from './routes/esri/rest.js';
 import { St } from './source.tracer.js';
-import { spriteGet } from './routes/sprites.js';
-import { fontGet, fontList } from './routes/fonts.js';
 
 const app = new Router();
 
-app.get('ping', Ping);
-app.get('health', Health);
-app.get('version', Version);
 app.get('tiles', Tiles);
-app.get('imagery', Imagery);
-app.get('esri', Esri);
 
-let slowTimer: NodeJS.Timer | null = null;
 export async function handleRequest(req: LambdaHttpRequest): Promise<LambdaHttpResponse> {
-  // Reset the request tracing
-  St.reset();
+  const apiKey = Router.apiKey(req);
+  if (apiKey != null) {
+    const apiKeyHash = createHash('sha256').update(apiKey).digest('base64');
+    req.set('api', apiKeyHash);
+  }
+  return await app.handle(req);
+}
 
-  // Warn if a request takes more than 10 seconds to process
-  if (slowTimer) clearTimeout(slowTimer);
-  slowTimer = setTimeout(() => req.log.warn(req.logContext, 'Lambda:Slow'), 10_000);
-  slowTimer.unref();
+export const handler = lf.http(LogConfig.get());
 
+handler.router.hook('request', (req) => {
   req.set('name', 'LambdaTiler');
-  try {
-    const apiKey = Router.apiKey(req);
-    if (apiKey != null) {
-      const apiKeyHash = createHash('sha256').update(apiKey).digest('base64');
-      req.set('api', apiKeyHash);
-    }
-    const ret = await app.handle(req);
+  // Reset the request tracing before every request
+  St.reset();
+});
 
+handler.router.hook('response', (req) => {
+  if (St.requests.length > 0) {
     // TODO this could be relaxed to every say 5% of requests if logging gets too verbose.
     req.set('requests', St.requests.slice(0, 100)); // limit to 100 requests (some tiles need 100s of requests)
     req.set('requestCount', St.requests.length);
-
-    return ret;
-  } finally {
-    if (slowTimer) clearTimeout(slowTimer);
-    slowTimer = null;
   }
-}
+});
 
-export const handler = lf.http(LogConfig.get());
+handler.router.get('/v1/ping', Ping);
+handler.router.get('/v1/health', Health);
```

Comment:
Do we also need to update the cdk code healthCheck which listen to the `'/health'` endpoint?

Reply:
hmm yeah I wonder how this works actually, `/v1/ping` works from the website eg https://basemaps.linz.govt.nz/v1/ping but its only registered as `/ping` 🤯